internal/resources/grafana/resource_dashboard_public.go

```diff
@@ -0,0 +1,145 @@
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourcePublicDashboard() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+Manages Grafana public dashboards.
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/dashboards/dashboard-public/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/dashboard_public/)
+`,
+
+		CreateContext: CreatePublicDashboard,
+		ReadContext:   ReadPublicDashboard,
+		UpdateContext: UpdatePublicDashboard,
+		DeleteContext: DeletePublicDashboard,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				Description: "The unique identifier of a public dashboard. " +
+					"It's automatically generated if not provided when creating a public dashboard. ",
+			},
+			"dashboard_uid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of the original dashboard.",
+			},
+			"access_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				Description: "A public unique identifier of a public dashboard. This is used to construct its URL. " +
+					"It's automatically generated if not provided when creating a public dashboard. ",
+			},
+			"time_selection_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Optional. Set to `true` to enable the time picker in the public dashboard. The default value is `false`.",
+			},
+			"is_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Optional. Set to `true` to enable the public dashboard. The default value is `false`.",
+			},
+			"annotations_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Optional. Set to `true` to show annotations. The default value is `false`.",
```

Comment:
No need to mark as Optional, the docs will put them in an optional section
```suggestion
				Description: "Set to `true` to show annotations. The default value is `false`.",
```